includes/api/class-urlslab-api-security.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+class Urlslab_Api_Security extends Urlslab_Api_Table {
+	const SLUG = 'security';
+
+	public function register_public_routes() {
+		$base = '/' . self::SLUG;
+
+		register_rest_route(
+			self::NAMESPACE,
+			$base . '/report_csp',
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'report_csp' ),
+					'permission_callback' => array(
+						$this,
+						'report_permissions_check',
+					),
+					'args'                => array(),
+				),
+			)
+		);
+	}
+
+	public function register_routes() {
+		$base = '/' . self::SLUG;
+
+		register_rest_route( self::NAMESPACE, $base . '/', $this->get_route_get_items() );
+		register_rest_route( self::NAMESPACE, $base . '/count', $this->get_count_route( array( $this->get_route_get_items() ) ) );
+
+		register_rest_route(
+			self::NAMESPACE,
+			$base . '/delete',
+			array(
+				array(
+					'methods'             => WP_REST_Server::ALLMETHODS,
+					'callback'            => array( $this, 'delete_items' ),
+					'permission_callback' => array(
+						$this,
+						'delete_item_permissions_check',
+					),
+					'args'                => array(),
+				),
+			)
+		);
+
+		register_rest_route(
+			self::NAMESPACE,
+			$base . '/delete-all',
+			array(
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_all_items' ),
+					'permission_callback' => array(
+						$this,
+						'delete_item_permissions_check',
+					),
+					'args'                => array(),
+				),
+			)
+		);
+	}
+
+
+	private function get_route_get_items(): array {
+		return array(
+			'methods'             => WP_REST_Server::CREATABLE,
+			'callback'            => array( $this, 'get_items' ),
+			'args'                => $this->get_table_arguments(),
+			'permission_callback' => array(
+				$this,
+				'get_items_permissions_check',
+			),
+		);
+	}
+
+	/**
+	 * @param WP_REST_Request $request
+	 *
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$rows = $this->get_items_sql( $request )->get_results();
+
+		if ( is_wp_error( $rows ) ) {
+			return new WP_Error( 'error', __( 'Failed to get items', 'urlslab' ), array( 'status' => 400 ) );
+		}
+
+		foreach ( $rows as $row ) {
+			$row->blocked_url_id = (int) $row->blocked_url_id;
+		}
+
+		return new WP_REST_Response( $rows, 200 );
+	}
+
+
+	public function report_permissions_check( $request ) {
+		return Urlslab_User_Widget::get_instance()->is_widget_activated( Urlslab_Widget_Security::SLUG );
+	}
+
+	public function report_csp( $request ) {
+		$json = json_decode( $request->get_body(), true );
+		if ( ! isset( $json['csp-report']['violated-directive'] ) || ! isset( $json['csp-report']['blocked-uri'] ) ) {
+			return new WP_Error( 'invalid_json', 'Invalid JSON', array( 'status' => 400 ) );
+		}
+
+		try {
+			$url = new Urlslab_Url( $json['csp-report']['blocked-uri'], true );
```

Comment:
we need to sanitize the input
https://developer.wordpress.org/reference/functions/sanitize_url/